изменения от  19.01.21

SergeyBersenev98 opened the pull request.

SergeyBersenev98 committed:
first commit

rah-emil commented on src/js/App.vue:
```diff
@@ -0,0 +1,20 @@
+</template>
+
+<script>
+  import MatchesData from './components/MatchesData.vue'
```
Внутри компонентов не обязательно указывать расширение .vue (даже не нужно)

rah-emil commented on src/js/components/HttpRequests.js:
```diff
@@ -0,0 +1,16 @@
+import Betting from '../../libs/betting.js';
+
+export const getDatafromServer = () => new Promise((resolve, reject) => {
```
Не ясно, почему helper-функция находится в папке компонентов и имеет довольно широкое название `HttpRequests.js`

rah-emil replied:
Также имеется слишком общее название `getDatafromServer` - какие data, с какого сервера? Метод же отвечает только за получение матчей :)

rah-emil commented on src/js/components/HttpRequests.js:
```diff
@@ -0,0 +1,16 @@
+    .then((token) => {
+      betting.get_matches({ lng: 'ru', count: 15 })
+        .then((matches) => resolve(matches))
+        .fail((err) => reject(err));
```
метод `fail` не будет работать. Необходимо вызывать `catch()`

rah-emil commented on src/js/components/HttpRequests.js:
```diff
@@ -0,0 +1,16 @@
+export const getDatafromServer = () => new Promise((resolve, reject) => {
+  const betting = new Betting();
+
+  betting.auth({
```
Нет обработки ошибки для метода `auth()`

rah-emil commented on src/js/components/MatchesData.vue:
```diff
@@ -0,0 +1,71 @@
+    name: 'MatchesData',
+    data(){
+      return{
+        unfilteredMatchesArray: [],
```
Очень не хорошо, что мы создаём 2 массива для данных

rah-emil replied:
Мы обязаны соблюдать SOLID и DRY принципы

rah-emil commented on src/js/components/MatchesData.vue:
```diff
@@ -0,0 +1,71 @@
+      return{
+        unfilteredMatchesArray: [],
+        filteredMatchesArray: [],
+        searchResults: 0,
```
Мы не должны создавать новую реактивную переменную ради такой мелочи и постоянно следить за количеством матчей. Для этого у нас существует конструкция `array.length`

rah-emil commented on src/js/components/MatchesData.vue:
```diff
@@ -0,0 +1,71 @@
+      }
+    },
+    mounted() {
+      this.isLoading = true
```
Допускается, что `then` у нас никогда не выполнится. Тогда у нас будет бесконечная загрузка

rah-emil commented on src/js/components/MatchesData.vue:
```diff
@@ -0,0 +1,71 @@
+      MatchesFilter,
+    },
+    methods: {
+      getUsersInput(usersInput){
```
Мелочный метод. Такой лучше сразу указывать внутри `<template>`

rah-emil commented on src/js/components/MatchesData.vue:
```diff
@@ -0,0 +1,71 @@
+        this.filteredMatchesArray = array
+      },
+
+      refreshData(){
```
В данном методе у нас, по сути, идёт копипаст. Дубль кода. Это нарушает принцип DRY

rah-emil commented on src/js/components/MatchesData.vue:
```diff
@@ -0,0 +1,71 @@
+        })
+      },
+
+      getSearchResults(resultsCount){
```
Метод не нужен так как есть `array.length`

rah-emil commented on src/js/components/MatchesFilter.vue:
```diff
@@ -0,0 +1,82 @@
+      <!-- /.search__input -->
+      <!-- /.search__filter -->
+      <button class="search__refresh" @click="refreshData">
+        <span class="iconify" data-icon="ci:refresh-02" style="color: white;"></span>
```
`style="color: white;"` тут лишний

rah-emil commented on src/js/components/MatchesFilter.vue:
```diff
@@ -0,0 +1,82 @@
+      <!-- /.search__refresh -->
+    </div>
+    <!-- /.search -->
+    <div v-bind:class="[searchResults ? 'search__report_visable' : 'search__report_unvisable']" >
```
1. `v-bind` всегда должны опускать
2. Класс `.search__report_unvisable` лишний

rah-emil commented on src/js/components/MatchesFilter.vue:
```diff
@@ -0,0 +1,82 @@
+    </div>
+    <!-- /.search -->
+    <div v-bind:class="[searchResults ? 'search__report_visable' : 'search__report_unvisable']" >
+      Результатов по запросу: {{searchResults}} 
```
Опять же: `array.length`

rah-emil commented on src/js/components/MatchesFilter.vue:
```diff
@@ -0,0 +1,82 @@
+        filteredMatchesArray: [],
+      }
+    },
+    props: ["unfilteredMatchesArray", "queryOpponent", "newQueryOpponent", "usersInput", "searchResults"],
```
Если переменных много, то указываем формат:
```javascript
props: {
  unfilteredMatchesArray: Array,
  queryOpponent: String,
  // ... и так далее
}
```

rah-emil commented on src/js/components/MatchesFilter.vue:
```diff
@@ -0,0 +1,82 @@
+    },
+    mounted(){},
+    methods: {
+      filterMatchesArray(){
```
Лишний метод. У нас должна идти обработка матчей в свойстве `computed`

rah-emil commented on src/js/components/MatchesList.vue:
```diff
@@ -0,0 +1,105 @@
+    <div v-if="!isLoading">
+      <div v-if="filteredMatchesArray.length" class="table">
+        <ul>
+          <li v-for="match in filteredMatchesArray" :key="match.constEventId">
```
Матч должен быть отдельным компонентом

rah-emil commented on src/js/components/MatchesList.vue:
```diff
@@ -0,0 +1,105 @@
+              <div v-if="match.hasVideo" class="table__translation_able">
+                <a :href="`${match.link}`">
+                  <span class="iconify" data-icon="el:play-alt"></span>
+                  {{ "Трансляция"}}
```
В фигурных скобках мы указываем только переменные, а строки без них

rah-emil commented on src/js/components/MatchesList.vue:
```diff
@@ -0,0 +1,105 @@
+                <!-- /a -->
+              </div>
+              <!-- /.table_is-translation-able -->   
+              <div v-if="!match.hasVideo" class="table__translation_disable">
```
это должен быть `v-else`

rah-emil commented on src/js/components/MatchesList.vue:
```diff
@@ -0,0 +1,105 @@
+</script>
+
+
+<style scoped>
```
Стили общие для загрузки, поэтому следует их указывать в scss в папке styles

rah-emil commented on src/styles/components/_content.scss:
```diff
@@ -1,23 +1,146 @@
+  font-size: 14px;
+}
+
+.versus{
```
Требует отдельный SCSS компонент versus

rah-emil commented on src/styles/components/_content.scss:
```diff
@@ -1,23 +1,146 @@
+  margin: auto
+}
+
+.table__translation{
```
Требует отдельный SCSS компонент table

rah-emil commented on src/styles/components/_content.scss:
```diff
@@ -1,23 +1,146 @@
+  font-family: 'PT Sans', sans-serif;
+}
+
+.search{
```
Требует отдельный SCSS компонент search

rah-emil commented on src/styles/components/_content.scss:
```diff
@@ -1,23 +1,146 @@
+  margin-bottom: 20px;      //to avoid text cutting
+}
+
+.loading{
```
Требует отдельный SCSS компонент loading

rah-emil commented on src/styles/components/_content.scss:
```diff
@@ -1,23 +1,146 @@
+  color: white;
+}
+
+.not-found{
```
Требует отдельный SCSS компонент not-found

rah-emil commented:
PullRequest будет принят после исправления всех указанных замечаний (☞ﾟヮﾟ)☞

SergeyBersenev98 committed:
Fixed some defect

Separate css-style file in modules. Create filter method. Delete not-necessary variables and arrays. Add data-format, add score. Create new component Match.vue

rah-emil commented on src/js/components/Match.vue:
```diff
@@ -0,0 +1,52 @@
+<template>
+ <ul>
+    <li v-for="match in matches" :key="match.constEventId">
```
Необходимо создать отдельный компонент для матча

rah-emil commented on src/js/components/Match.vue:
```diff
@@ -0,0 +1,52 @@
+ <ul>
+    <li v-for="match in matches" :key="match.constEventId">
+      <div class="table__time">
+          {{Math.floor(match.timeSec/60)}}
```
Это не перевод времени)
Нужно сделать через `Date`, либо moment.js

Детали: https://yandex.ru/search/?text=js+unix+time+to+date&lr=213

rah-emil commented on src/js/components/Match.vue:
```diff
@@ -0,0 +1,52 @@
+                {{match.fullScore.sc1}} : {{match.fullScore.sc2}}
+              </div>
+              <!-- /.versus__score -->       
+              <img :src="`https://nimblecd.com/sfiles/logo_teams/${match.imageOpponent2}`" class="versus__team-img"></img>
```
Реализовать ссылку через метод с параметром

rah-emil commented on src/js/components/Match.vue:
```diff
@@ -0,0 +1,52 @@
+                {{match.opponent1NameLocalization}}
+              </div>
+              <!-- /.versus__team-first-localization --> 
+              <img :src="`https://nimblecd.com/sfiles/logo_teams/${match.imageOpponent1}`" class="versus__team-img"></img>
```
Реализовать ссылку через метод с параметром

rah-emil commented on src/js/components/Match.vue:
```diff
@@ -0,0 +1,52 @@
+            <!-- /.versus -->
+            <div class="table__translation">
+              <div v-if="match.hasVideo" class="table__translation_able">
+                <a :href="`${match.link}`">
```
Добавить `target="_blank"`, чтобы по клику не покидали сайт

rah-emil commented on src/js/components/Match.vue:
```diff
@@ -0,0 +1,52 @@
+                <!-- /a -->
+              </div>
+              <!-- /.table_is-translation-able -->   
+              <div v-if="!match.hasVideo" class="table__translation_disable">
```
Опять же, `v-else`

rah-emil commented on src/js/components/MatchesData.vue:
```diff
@@ -1,71 +1,65 @@
+      matchesFilter() {
+        let filteredMatchesArray = [];
+        if(this.usersInput !== ""){
+             filteredMatchesArray = this.matches.filter((name) => {
```
Не корректно указываем название перебираемого объекта. В матчах мы перебираем в методе `filter` не объект имени, а объект матча

rah-emil commented on src/js/components/MatchesData.vue:
```diff
@@ -1,71 +1,65 @@
-                   :unfilteredMatchesArray="unfilteredMatchesArray"/>
-    <MatchesList :filteredMatchesArray="filteredMatchesArray"
-                 :isLoading="isLoading"/>    
+    <MatchesFilter :resultsForSearch="matchesFilter.length"
```
Подразумевается, что ты возвращаешь тут только целое число. Соответственно лучше указывать название переменной не `resultsForSearch`, а `countFilteredMatches`, то есть - КОЛИЧЕСТВО ОТФИЛЬТРОВАННЫХ МАТЧЕЙ. Следим за этим :)

rah-emil commented on src/js/components/MatchesFilter.vue:
```diff
@@ -3,21 +3,23 @@
-      Результатов по запросу: {{searchResults}} 
-    </div>
+    <div class="search__report">
+      <div v-if="resultsForSearch" class='search__report_isHasReports' >
```
Тут не обязательно указывать условие, так как по условию мы тут показываем кол-во выводимых матчей, соответственно можем показать, что их 0, если не найдено. По макету так и показано

rah-emil commented on src/js/components/MatchesFilter.vue:
```diff
@@ -28,55 +30,21 @@
-      value() { 
-        this.$emit('getUsersInput', this.value);
+      usersInput() { 
+        this.$emit('getUsersInput', this.usersInput);
```
Для простоты понимания можно просто указать событие `input`, а не `getUsersInput`

rah-emil commented on src/js/components/MatchesList.vue:
```diff
@@ -1,53 +1,7 @@
-      </div>
-      <!-- /.table -->
-      <div v-else class="not-found">
+      <div v-if="matches.length === 0" class="not-found">
```
Создай компонент `NotFoundMatches.vue`

rah-emil replied:
Это удобно, потому что:
- мы чётко "в лоб" разделяем код, и поддерживать его проще
- имеется возможность пере использования кода с соблюдением принципа DRY

rah-emil commented on src/js/components/MatchesList.vue:
```diff
@@ -58,6 +12,10 @@
+      <!-- /.table -->
     </div>
     <!-- /if -->
     <div v-else>
```
Аналогично с компонентов "не найдено". Нужно создать компонент загрузки

rah-emil commented:
Коммент по поводу всех файлов scss.
Отсутствует вложенность(

SergeyBersenev98 committed:
Create Loading component and NotFoundMathces

SergeyBersenev98 committed:
scss fixed

rah-emil commented:
Лично объяснил остаточные работы

SergeyBersenev98 committed:
create timer and fixed time

rah-emil commented on src/js/components/Match.vue:
```diff
@@ -40,13 +40,34 @@
 </template>
 
 <script>
+    import moment from 'moment'
```
Не делать так. реализуй через `vue-moment`, а не напрямую через `moment`

SergeyBersenev98 committed:
use vue-moment instead of import moment
